FeedStoreChallenge/CoreDataFeedStore/CoreDataFeedStore.swift

```diff
@@ -0,0 +1,130 @@
+//
+//  CoreDataFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Edgar Hirama on 08/03/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import CoreData
+
+public class CoreDataFeedStore: FeedStore {
+
+	enum CoreDataError: Error {
+		case unableToFindModel
+		case unableToLoad(Error)
+	}
+
+	private static let resourceName: String = "FeedStore"
+
+	private let persistentContainer: NSPersistentContainer
+	private let context: NSManagedObjectContext
+
+	public init(storeURL: URL) throws {
+		self.persistentContainer = try CoreDataFeedStore.loadPersistendContainer(storeURL: storeURL)
+		self.context = persistentContainer.newBackgroundContext()
+	}
+
+	private static func loadPersistendContainer(storeURL: URL) throws -> NSPersistentContainer {
+		let bundle = Bundle(for: CoreDataFeedStore.self)
+		guard let bundleURL = bundle.url(forResource: CoreDataFeedStore.resourceName, withExtension: "momd"),
+			  let model = NSManagedObjectModel(contentsOf: bundleURL) else { throw CoreDataError.unableToFindModel }
+
+		let container = NSPersistentContainer(name: CoreDataFeedStore.resourceName, managedObjectModel: model)
+		let description = NSPersistentStoreDescription(url: storeURL)
+		container.persistentStoreDescriptions = [description]
+
+		var loadError: Error?
+		container.loadPersistentStores { _, error in
+			loadError = error
+		}
+		if let error = loadError {
+			throw CoreDataError.unableToLoad(error)
+		}
+		return container
+	}
+
+	private func deleteCurrentCacheIfNeeded(in context: NSManagedObjectContext) {
+		let fetchRequest: NSFetchRequest<LocalCache> = NSFetchRequest(entityName: LocalCache.className())
+		if let fetchResult = try? context.fetch(fetchRequest), let existingFeed = fetchResult.first {
```

Comment:
This line can be simplified in a single statement:

```suggestion
		if let existingFeed = try? context.fetch(fetchRequest).first {
```